resources.py

```diff
@@ -395,6 +395,27 @@ def wait_for_port(self, port, state, timeout=30):
 
         raise Timeout(f"Timed out waiting for {self}:{port} to be {state}")
 
+    @with_trigger('server.wait-for-nested-vm-boot')
+    def wait_for_nested_vm_boot(self, vm_name, path, text, timeout=30):
```

Comment:
This could be made more generic: `wait_for_text_in_file`, then pass a `path` and a `text`. I think that would clarify the test further up as well (otherwise `wait_for_nested_vm_boot` is quite opaque).